File: 3737-count-subarrays-with-majority-element-i/solution.py
class Solution:
    def countSubarrays(self, nums: list[int], target: int) -> int:
        """
        Calculates the number of subarrays where target is the majority element.
        Uses an O(N^2) approach suitable for N <= 1000.
        """
        n = len(nums)
        valid_subarrays = 0
        
        for i in range(n):
            count_target = 0
            for j in range(i, n):
                if nums[j] == target:
                    count_target += 1
                
                length = j - i + 1
                # Majority condition: count_target > length / 2
                # Equivalent to 2 * count_target > length
                if 2 * count_target > length:
                    valid_subarrays += 1
                    
        return valid_subarrays

File: 3737-count-subarrays-with-majority-element-i/solution.cpp
#include <vector>

class Solution {
public:
    /**
     * Iterates through all possible subarrays and checks the majority condition.
     * The condition 'count of target > length / 2' is equivalent to
     * (2 * count_target) > length.
     */
    int countSubarrays(std::vector<int>& nums, int target) {
        int n = nums.size();
        int validSubarrays = 0;

        for (int i = 0; i < n; ++i) {
            int countTarget = 0;
            for (int j = i; j < n; ++j) {
                if (nums[j] == target) {
                    countTarget++;
                }
                
                int length = j - i + 1;
                // Majority condition: count > length / 2
                // Multiply by 2 to avoid floating point issues: 2 * count > length
                if (2 * countTarget > length) {
                    validSubarrays++;
                }
            }
        }
        
        return validSubarrays;
    }
};